src/context/context.ts

```diff
@@ -67,9 +67,12 @@ export class Context {
   }
 }
 
-export function readProperty (obj: Scope, key: string) {
+export function readProperty (obj: Scope, key: string): any {
   if (isNil(obj)) return obj
   obj = toLiquid(obj)
+  if (obj instanceof Promise) {
+    return obj.then(resolvedObj => readProperty(resolvedObj, key))
```

Comment:
I think we need await the `obj` when calling `readProperty` rather than inside `readProperty`, it's passed in by the caller after all.

If we also need to support `toLiquid` returning a Promise, we should make it explicit by `obj = yield toLiquid(obj)` and a corresponding test case. (I'm not sure if you want this, it's up to you).

Reply:
But don't we still need to await inside readProperty if the promise was an object with properties?

Reply:
I believe not, `readProperty` is not recursive and only called from `getFromScope`. We can yield the obj before pass into the `readProperty`, like:

```
readProperty(yield scope, path)
```

Reply:
Since i don't see a way to implement these changes without introducing breaking changes, i am going to close this PR.

Appreciated your time